public/js/views/management/history.jsx

```diff
@@ -1,18 +1,90 @@
-import React, { Component } from 'react';
+import React, { Component, PropTypes } from 'react';
+import JsonTable from 'react-json-table';
 
 import { gettext, setTitle } from 'utils';
+import * as products from 'products';
+import Spinner from 'components/spinner';
 
 
 export default class History extends Component {
 
+  static propTypes = {
+    getUserTransactions: PropTypes.func.isRequired,
+    transactions: PropTypes.array.isRequired,
+  };
+
   componentDidMount() {
     setTitle(gettext('Receipts and history'));
+    this.props.getUserTransactions();
+  }
+
+  showReceipt(e, transactionId) {
+    e.preventDefault();
+    console.log('TODO: show receipt for transaction', transactionId);
+  }
+
+  selectSeller(e) {
+    e.preventDefault();
+    console.log('TODO: filter transactions by seller', e.target);
+  }
+
+  renderContent() {
+
+    var columns = [
+      {
+        key: 'created',
+        label: gettext('Date'),
+        cell: (item, columnKey) => (
+            // TODO: parse and format date like:
+            // "2015-08-07T14:53:23.966" : 'Aug 7, 2015'
+            item[columnKey]
+        ),
+      },
+      {
+        key: 'seller',
+        label: (
+          <select onChange={this.selectSeller}>
+            <option value="">
+              {gettext('View transactions from all companies')}
+            </option>
+            <option value="example-seller">Example seller</option>
+          </select>
+        ),
+        cell: (item) => (
+          // TODO: get localized seller name, not just English.
+          products.get(item.transaction.seller_product.public_id).seller.name.en
+        ),
+      },
+      {
+        key: 'amount',
+        label: gettext('Amount'),
+        cell: (item) => (
+          // TODO: display currency with symbol (when possible)
+          <a onClick={(e) => this.showReceipt(e, item.resource_pk)}
+             href="#">{item.transaction.currency} {item.transaction.amount}</a>
+        ),
+      },
+    ];
+
+    // TODO: this is currently showing all transactions but that might be
+    // weird when it includes kinds other than subscription_charged_successfully
+
+    if (this.props.transactions === null) {
+      return <Spinner/>;
+    } else {
+      if (this.props.transactions.length) {
+        return <JsonTable rows={this.props.transactions} columns={columns} />;
+      } else {
+        return <p>{gettext("You haven't purchased anything yet.")}</p>;
+      }
+    }
   }
 
   render() {
     return (
       <div>
-        <h1>{gettext('Receipts and history')}</h1>
+        <h1>{gettext('Transaction Receipts & History')}</h1>
```

Comment:
Everything tends to only upper case the first word of the title. So this should probably be:

`Transaction receipts & history`

Does 'and' not fit? I'll make the sidebar more flexible in due course but as a quick fix you could make it wider if you want to use `and` instead of `&`